### ISSUE TITLE
Broker interface should be programmatically enforced

### ISSUE THREAD

**Issue by jmcph4** [kieran-mackle/AutoTrader#55]
**Is your feature request related to a problem? Please describe.**
A common desire is to extend AutoTrader's functionality to new exchanges (either specific instances of exchanges or entire classes, such as DEXes). In order to do this, new implementations must conform to the [broker interface](https://autotrader.readthedocs.io/en/latest/broker/broker-interface.html). Unfortunately, this interface, despite having [very real](https://autotrader.readthedocs.io/en/latest/broker/broker-interface.html#methods) expectations on compliant implementations, is not enforced anywhere in the current codebase.

Python purists will argue this is not Pythonic as it attempts to [sidestep ducktyping](https://stackoverflow.com/a/2124415/8028639); however, there is [clearly confusion](https://github.com/kieran-mackle/AutoTrader/issues/47) surrounding exactly what a broker must look like.

**Describe the solution you'd like**
At a high level, programmatic enforcement of the broker interface (which exists regardless of enforcement method). Actually achieving this in Python is sadly a best-effort endeavour though.

The least insane solution is very likely an [abstract base class](https://docs.python.org/3/library/abc.html). Specifically, by defining [abstract methods](https://realpython.com/python-interface/#using-abstract-method-declaration). Something like the following should suffice:

```python
class Broker(metaclass=abc.ABCMeta):
    @classmethod
    def __subclasshook__(cls, subclass):
        return (hasattr(subclass, 'get_NAV') and 
                callable(subclass.get_NAV) and 
                hasattr(subclass, 'get_balance') and 
                callable(subclass.get_balance) and
                hasattr(subclass, 'place_order') and 
                callable(subclass.place_order) and
                hasattr(subclass, 'get_orders') and
                callable(subclass.get_orders) and
                hasattr(subclass, 'cancel_order') and
                callable(subclass.cancel_order) and
                hasattr(subclass, 'get_trades') and 
                callable(subclass.get_trades) and
                hasattr(subclass, 'get_positions') and
                callable(subclass.get_positions) or
                NotImplemented)

    @abc.abstractmethod
    def get_NAV(self) -> float:
       pass
       
    # etc. ...
```

**Describe alternatives you've considered**
Other methods like [subclass hooks](https://realpython.com/python-interface/#using-__subclasshook__) but these seem ugly and complicated (admittedly my Python knowledge may be lacking here and these might be The One True Way(TM)).

**Additional context**
None.

**Comment by jmcph4** [comment]
In hindsight, `Broker` is an unfortunate name for that snippet as it collides with existing conventions:

https://github.com/kieran-mackle/AutoTrader/blob/9478b6253bcd9456ad3dee99cd8b961950196ba3/autotrader/brokers/dydx/broker.py#L11-L14

**Comment by kieran-mackle** [comment]
Great suggestion, it has been an issue on my radar for a while and is definitely something which should be improved. Will bump it up the priority list.

**Comment by jmcph4** [comment]
Related: #7

**Comment by kieran-mackle** [comment]
Implemented with #60
